Exclude the vendor directory from code scanning

niik opened the pull request:
<!--
What GitHub Desktop issue does this PR address? (for example, #1234)
-->


## Description

We're getting Code Scanning issues reported by our [vendored yarn](https://github.com/desktop/desktop/blob/d9279df6ff51f96d34e4a491390e326a26e074cd/vendor/). There's no risk of that file creating security vulnerabilities in GitHub Desktop so in order to reduce the noise we're gonna ignore it.

niik committed:
Exclude the vendor directory from code scanning

rafeca commented:
Doesn't seem like the code scanning skipped the yarn vendor file (at least based on the open reports for this branch: https://github.com/desktop/desktop/security/code-scanning?query=ref%3Arefs%2Fheads%2Fcodeql-config).

rafeca commented on .github/codeql/codeql-config.yml:
```diff
@@ -0,0 +1,4 @@
+name: 'GitHub Desktop CodeQL config'
+
+paths-ignore:
+  - '/vendor/'
```
Maybe this needs to be something like this?

```suggestion
  - './vendor/'
```

niik replied:
I have no clue, it  was copied straight from https://help.github.com/en/github/finding-security-vulnerabilities-and-errors-in-your-code/configuring-code-scanning but https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#onpushpull_requestpaths seems to suggest it's supposed to be a pattern. Maybe something like this?

```suggestion
  - '/vendor/'
  +'/vendor/yarn*.js'
```

niik replied:
Actually, looking at https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet it seems like the patterns are relative to the repository root.

```suggestion
  - '/vendor/'
  - 'vendor/yarn*.js
```

niik committed:
Update .github/codeql/codeql-config.yml

niik committed:
Face, meet palm

niik committed:
Only run on pushes to development

We enforce all changes going through PRs so the current approach means we'll always run it twice

niik committed:
How about this then?

niik committed:
Last attempt before I toss this all out the window

niik commented:
I've also added a config option here so that we only build on pushes to `development`. Since development is a protected branch that we can only push to via PRs it's redundant to build for both pushes and PRs

rafeca reviewed (approved):
![security](https://user-images.githubusercontent.com/408035/85569563-f0099000-b632-11ea-8cfc-142323f60fc3.gif)

niik commented:
Gonna merge this early to clean up our code scanning results

mathieuedwards commented on .github/workflows/codeql.yml:
```diff
@@ -2,6 +2,8 @@ name: 'Code scanning - action'
 on:
   push:
+    branches:
+      - development
```
.github/workflows/codeql.yml

mathieuedwards replied:
.github/workflows/codeql.yml
